src/bp/nlu-core/intents/exact-matcher.ts

```diff
@@ -0,0 +1,41 @@
+import { MLToolkit } from 'botpress/sdk'
+import _ from 'lodash'
+import { NONE_INTENT } from 'nlu-core/training-pipeline'
+import { Intent } from 'nlu-core/typings'
+import Utterance, { UtteranceToStringOptions } from 'nlu-core/utterance/utterance'
+
+export const EXACT_MATCH_STR_OPTIONS: UtteranceToStringOptions = {
+  lowerCase: true,
+  onlyWords: true,
+  slots: 'keep-value', // slot extraction is done in || with intent prediction
```

Comment:
That is for training time where slots are tagged. comment is not necessary

Reply:
Yeah, we use to replace slot by value in the exact-match, but it had no effect as when exact matcher runs, slots arent extracted yet.

This is true for both training and prediction.

The comment has some history, but I can remove it.